lib/ecto/adapters/mysql/connection.ex

```diff
@@ -11,7 +11,13 @@ if Code.ensure_loaded?(Mariaex.Connection) do
 
     def connect(opts) do
       opts = Keyword.update(opts, :port, @default_port, &normalize_port/1)
-      Mariaex.Connection.start_link(opts)
+      case Mariaex.Connection.start_link(opts) do
+        {:ok, pid} ->
+          repo = Keyword.fetch!(opts, :repo)
+          repo.after_connect(pid)
```

Comment:
Yeah except we only need to catch the `Task.await/2` exits:
```elixir
catch
   :exit, {:timeout, [Task, :await, [%Task{pid: task_pid}, _]}} ->
      Process.exit(task_pid, :kill)
      {:error, :timeout}
   :exit, {reason, {Task, :await, _}} ->
      {:error, reason}
```
I messed up the catch, its not `{class, reason}`, its `class, reason`. That's what happens when writing and refactoring in a github comment :).

Notice that `timeout` is a special case because the task is still alive. Then we can disconnect the connection as you did.